src/java.base/share/classes/java/security/PEMDecoder.java

```diff
@@ -0,0 +1,486 @@
+        Objects.requireNonNull(str);
+        DEREncodable de;
+        try {
+            return decode(new ByteArrayInputStream(
+                str.getBytes(StandardCharsets.UTF_8)));
+        } catch (IOException e) {
+            // With all data contained in the String, there are no IO ops.
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    /**
+     * Decodes and returns a {@link DEREncodable} from the given
+     * {@code InputStream}.
+     *
+     * <p>This method reads the {@code InputStream} until PEM data is
+     * found or until the end of the stream.  It can be called repeatedly
+     * on the {@code InputStream} to decode additional PEM data.
+     * The {@code InputStream} read pointer will remain at the end of
+     * the PEM footer unless an IOException occurs.
+     *
+     * <p> This method returns a Java API cryptographic object,
+     * such as a {@code PrivateKey}, if the PEM type is supported.
+     * Any non-PEM data preceding the PEM header is ignored by the decoder.
+     * If no cryptographic object is found, a {@link PEMRecord} will be
+     * returned containing the type identifier and Base64-encoded data.
+     * Any non-PEM data preceding the PEM header will be stored in
+     * {@code leadingData}.
+     *
+     * <p> If no PEM data is found, an {@code IllegalArgumentException} is
+     * thrown.
+     *
+     * @param is InputStream containing PEM data
+     * @return a {@code DEREncodable}
+     * @throws IOException on IO error with the {@code InputStream}
+     * @throws EOFException the end of {@code InputStream} has been
+     * unexpectedly reached.
+     * @throws IllegalArgumentException on error in decoding or no PEM data
+     * found
+     * @throws NullPointerException when {@code is} is null.
+     */
+    public DEREncodable decode(InputStream is) throws IOException {
+        Objects.requireNonNull(is);
+        PEMRecord pem = Pem.readPEM(is);
+        return decode(pem);
+    }
+
+    /**
+     * Decodes and returns a {@code DEREncodable} of the specified class from
+     * the given PEM string. {@code tClass} must extend {@link DEREncodable}
+     * and be an appropriate class for the PEM type.
+     *
+     * <p> This method reads the {@code String} until PEM data is found
+     * or the end of the {@code String} is reached.  If no PEM data is found,
+     * an {@code IllegalArgumentException} is thrown.
+     *
+     * <p> If the class parameter is {@code PEMRecord.class},
+     * a {@linkplain PEMRecord} is returned containing the
+     * type identifier and Base64 encoding. Any non-PEM data preceding
+     * the PEM header will be stored in {@code leadingData}.  Other
+     * class parameters will not return preceding non-PEM data.
+     *
+     * <p> Input consumed by this method is read in as
+     * {@link java.nio.charset.StandardCharsets#UTF_8 UTF-8}.
+     *
+     * <p> For all other class parameters, {@code IllegalArgumentException} is
```

Comment:
"For all other class parameters", you mean not `PEMRecord`? There is a paragraph in between (UTF-8 thingy) and users have forgotten what "other" means.